integrations/influxdb_metrics_dashboard/influxdb_metrics_dashboard.go

```diff
@@ -0,0 +1,389 @@
+	workspacePolicy := awsiam.NewPolicy(stack, jsii.String("influxdb-dashboard-workspace-policy"), &awsiam.PolicyProps{
+		Statements: &[]awsiam.PolicyStatement{
+			awsiam.NewPolicyStatement(&awsiam.PolicyStatementProps{
+				Actions: &[]*string{
+					jsii.String("timestream:DescribeEndpoints"),
+				},
+				Resources: &[]*string{
+					jsii.String("*"),
+				},
+			}),
+			awsiam.NewPolicyStatement(&awsiam.PolicyStatementProps{
+				Actions: &[]*string{
+					jsii.String("timestream:Select"),
+				},
+				Resources: &[]*string{
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/storage_bucket_series_num", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/go_memstats_mcache_inuse_bytes", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/boltdb_writes_total", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/go_memstats_alloc_bytes", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/http_write_request_count", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/go_memstats_frees_total", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/go_memstats_sys_bytes", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/qc_executing_duration_seconds", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+					jsii.String(fmt.Sprintf("arn:aws:timestream:%s:%s:database/%s/table/http_query_request_count", *stackProps.Env.Region, *stackProps.Env.Account, databaseName)),
+				},
+			}),
+		},
+	})
+
+	workspaceRole := awsiam.NewRole(stack, jsii.String("influxdb-dashboard-workspace-role"), &awsiam.RoleProps{
+		AssumedBy: awsiam.NewServicePrincipal(jsii.String("grafana.amazonaws.com"), nil),
+	})
+
+	workspacePolicy.AttachToRole(workspaceRole)
+
+	awsgrafana.NewCfnWorkspace(stack, jsii.String(grafanaWorkspaceName), &awsgrafana.CfnWorkspaceProps{
+		AccountAccessType:       jsii.String("CURRENT_ACCOUNT"),
+		AuthenticationProviders: &[]*string{jsii.String("AWS_SSO")},
+		PermissionType:          jsii.String("CUSTOMER_MANAGED"),
+		PluginAdminEnabled:      true,
+		RoleArn:                 workspaceRole.RoleArn(),
+		Name:                    jsii.String(grafanaWorkspaceName),
+	})
+
+	return stack, nil
+}
+
+func createLambdaResource(stack awscdk.Stack, stackProps awscdk.StackProps, databaseName string, grafanaWorkspaceName string, dashboardName string, timestreamDatasourceName string) (awscdk.Stack, error) {
```

Comment:
The Lambda function is executed after the CFN stack has been deployed, and sets up the Grafana data source and uploads the dashboard. I think it would be good to add some more info clarifying this in the README. Do you think I should also add it to the architecture diagram?

Reply:
I agree, it being mentioned in the README is good enough. Users should at least be aware that a Lambda function will be created.